### PR TITLE
[Merged by Bors] - Add helper methods for rotating `Transform`s

### PR DESCRIPTION
**Pull request by devil-ira**
# Objective
Users often ask for help with rotations as they struggle with `Quat`s.
`Quat` is rather complex and has a ton of verbose methods.

## Solution
Add rotation helper methods to `Transform`.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Ping @bitshifter for review.

**Comment by devil-ira** [comment]
CI failure is a little funny since it is *my* code giving a warning... but it's from a different PR that is already merged.
Anyways, new rust stable release, yay!

File: crates/bevy_transform/src/components/transform.rs [hunk 1]
```diff
@@ -195,20 +195,81 @@ impl Transform {
...
+    pub fn rotate_axis(&mut self, axis: Vec3, angle: f32) {
+        self.rotate(Quat::from_axis_angle(axis, angle));
+    }
+
+    /// Rotates this [`Transform`] on the X axis by `angle` (in radians).
+    ///
+    /// If this [`Transform`] has a parent, the axis is relative to the rotation of the parent.
+    #[inline]
+    pub fn rotate_x(&mut self, angle: f32) {
+        self.rotate_axis(Vec3::X, angle);
```
Comment (bitshifter) [inline crates/bevy_transform/src/components/transform.rs:217]: This is less efficient than using `Quat::from_rotation_x` as `Quat::from_axis_angle` needs to multiply the `Vec3` `axis` by the sine of the angle, where as `Quat::from_rotation_x` does not.

File: crates/bevy_transform/src/components/transform.rs [hunk 2]
```diff
@@ -195,20 +195,81 @@ impl Transform {
...
+    }
+
+    /// Rotates this [`Transform`] on the Y axis by `angle` (in radians).
+    ///
+    /// If this [`Transform`] has a parent, the axis is relative to the rotation of the parent.
+    #[inline]
+    pub fn rotate_y(&mut self, angle: f32) {
+        self.rotate_axis(Vec3::Y, angle);
```
Comment (bitshifter) [inline crates/bevy_transform/src/components/transform.rs:225]: Same comment as `rotate_x`

File: crates/bevy_transform/src/components/transform.rs [hunk 3]
```diff
@@ -195,20 +195,81 @@ impl Transform {
...
+    }
+
+    /// Rotates this [`Transform`] on the Z axis by `angle` (in radians).
+    ///
+    /// If this [`Transform`] has a parent, the axis is relative to the rotation of the parent.
+    #[inline]
+    pub fn rotate_z(&mut self, angle: f32) {
+        self.rotate_axis(Vec3::Z, angle);
```
Comment (bitshifter) [inline crates/bevy_transform/src/components/transform.rs:233]: Same comment as `rotate_y`.

File: crates/bevy_transform/src/components/transform.rs [hunk 4]
```diff
@@ -195,20 +195,81 @@ impl Transform {
...
-    /// Rotates the transform by the given rotation.
+    /// Rotates this [`Transform`] by the given rotation.
     #[inline]
     pub fn rotate(&mut self, rotation: Quat) {
         self.rotation = rotation * self.rotation;
     }
 
-    /// Rotates this [`Transform`] around a point in space.
-    /// If the point is a zero vector, this will rotate around the parent (if any) or the origin.
+    /// Rotates this [`Transform`] on the given `axis` by `angle` (in radians).
```
Comment (bitshifter) [inline crates/bevy_transform/src/components/transform.rs:204]: Wording wise I prefer `around` the given axis to `on` the given axis.

**Review by IceSentry — APPROVED** [review]
This is great, I've seen a lot of people struggling with quaternions.

**Pull request closed without merging**